instance/models/mixins/nomad_jobs.py

```diff
@@ -0,0 +1,83 @@
+# -*- coding: utf-8 -*-
+#
+# OpenCraft -- tools to aid developing and hosting free software projects
+# Copyright (C) 2015-2018 OpenCraft <contact@opencraft.com>
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+"""
+Mixin for instances that depend on Nomad jobs.
+"""
+
+# Imports #####################################################################
+
+import time
+
+from django.db import models
+
+
+# Classes #####################################################################
+
+class NomadTimeoutError(Exception):
+    """Exception indicating that a Nomad operation did not complete within the expected time."""
+
+
+class NomadJobMixin(models.Model):
+    """A mixin for an Instance subclass to launch Nomad jobs."""
+
+    class Meta:
+        abstract = True
+
+    def get_nomad_jobs(self): # pylint: disable=no-self-use
+        """Return an iterable of instance.nomad_client.NomadJob instances describing the job to run.
+
+        This method should be overwritten in subclasses.
+        """
+        return []
+
+    def provision_nomad_jobs(self, timeout=120):
```

Comment:
The jobs should come up in just a few seconds, so I don't foresee the need to ever tune this.  If it doesn't come up in two minutes, we can most definitely assume that something went wrong.

If we ever revisit this part (which I hope we don't before moving to Terraform), the first thing to improve would be to also check for any potential error conditions and error out immediately if one is detected instead of waiting for two minutes.